platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/TemporaryFileBackedOutputStream.java

```diff
@@ -83,12 +96,29 @@ public void close() throws IOException {
         }
         try {
             fileBackedOutputStream.close();
-            fileBackedOutputStream.reset();
+            reset();
         } finally {
             isClosed = true;
         }
     }
 
+    /**
+     * Reset fileBackedOutputStream and retry if it fails.
+     */
+    private void reset() throws IOException {
+        try {
+            OperationRunner.runWithRetry(MAX_RETRY_ATTEMPTS,
```

Comment:
You could give `runWithRetry` a simpler signature (Operation operation, Consumer<Exception> exceptionHandler) and then include default values in the `OperationRunner` class. It could make default uses of this class a little cleaner. You could also give it a fluent constructor and have some methods like `.retryAttempts(int attempts)`, `.initialSleep(int millis)` etc similar to the [DDF WaitCondition](https://github.com/codice/ddf/blob/975288e7570d1d5d0fc683cc4e8b643a100e01d6/distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/WaitCondition.java)